pairinteraction/CMakeLists.txt

```diff
@@ -155,14 +182,35 @@ if(WITH_GSL)
   target_compile_definitions(picomplex PUBLIC WITH_GSL)
 endif()
 
-find_package(MKL)
-if (MKL_FOUND)
-  target_include_directories(pireal SYSTEM PUBLIC ${MKL_INCLUDE_DIR})
-  target_include_directories(picomplex SYSTEM PUBLIC ${MKL_INCLUDE_DIR})
-  target_link_libraries(pireal PUBLIC ${MKL_LIBRARY})
-  target_link_libraries(picomplex PUBLIC ${MKL_LIBRARY})
+set(MKL_ARCH "intel64" CACHE STRING "MKL architecture")
+set(MKL_LINK "dynamic" CACHE STRING "MKL linkage")
+set(MKL_THREADING "sequential" CACHE STRING "MKL threading backend")
+set(MKL_INTERFACE "lp64" CACHE STRING "MKL interface")
+find_package(MKL CONFIG)
```

Comment:
Could we use MKL instead of OpenBLAS for Eigen if MKL is found (https://eigen.tuxfamily.org/dox/TopicUsingIntelMKL.html)? Maybe, one could gain some additional speedup [[1]](https://www.intel.com/content/www/us/en/developer/articles/technical/performance-comparison-of-openblas-and-intel-math-kernel-library-in-r.html).

Reply:
That's being done already (`-DBLA_VENDOR=Intel10_64lp_seq`) but only on the Debian build which is the only one that has MKL available and isn't deployed anywhere.

Reply:
Perfect (`EIGEN_USE_MKL_VML` is not set but does presumably not lead to additional performance gains because we do not make heavy use of the Eigen functions that would be affected by this flag, do you agree? ). It would be great if we could add a remark to the documentation so that advanced users can directly see how to substitute OpenBLAS by MKL, ideally, together with some benchmarking results. However, I see that this is a lot of additional effort. We could keep it in mind and do it once we clean up the installation instructions.

Reply:
I just pushed a commit that takes care of this. If MKL is found, it is used automatically with `EIGEN_USE_MKL_ALL`, otherwise LAPACK is searched normally and `EIGEN_USE_BLAS` and `EIGEN_USE_LAPACKE` are used.